#VRML_SIM R2022a utf8
WorldInfo {
}
Viewpoint {
  orientation -0.1391744145953113 0.1398744226220544 0.9803395474110995 1.5856552786293516
  position -0.017602020918130998 -2.8895795761312097 1.416880218819915
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
}
DEF ROBOT Robot {
  translation 0 0 0.04
  rotation 1 0 0 1.5708
  children [
    DEF EMITTER Emitter {
    }
    DEF RECEIVER Receiver {
    }
    DEF HINGE_COVER Solid {
      translation 0 0.03 -3.469446951953614e-18
      rotation 0 1 0 -1.5707953071795862
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0.6509803921568628 1
          }
          geometry Box {
            size 0.030000000000000002 0.019999999999999997 0.05
          }
        }
      ]
      name "hingeCover"
    }
    DEF BODY Shape {
      appearance PBRAppearance {
        baseColor 0.917647 0.145098 0.145098
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.2 0.05 0.08
      }
    }
    DEF WHEEL1 HingeJoint {
      jointParameters HingeJointParameters {
        position 6.723540149111039e-17
        axis 0 0 1
        anchor 0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint Solid {
        translation 0.060001004644818584 1.3392769184168344e-05 0.050000010543824816
        rotation 1.7671082538995992e-08 -1.7671248834016876e-08 0.9999999999999998 1.5708000430119498
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColor 0.305882 0.898039 0.25098
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
              subdivision 24
            }
          }
        ]
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL2 HingeJoint {
      jointParameters HingeJointParameters {
        position 6.670640318240083e-17
        axis 0 0 1
        anchor -0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint Solid {
        translation -0.060001095993954395 7.61633872322323e-07 0.05000000952472374
        rotation 1.7609654043190202e-08 -1.760982618388981e-08 0.9999999999999997 1.570799956989709
        children [
          USE WHEEL
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL3 HingeJoint {
      jointParameters HingeJointParameters {
        position 1.0324509589061627e-16
        axis 0 0 1
        anchor 0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation 0.06000100379909951 1.339265993453469e-05 -0.05000000738328558
        rotation -1.7671380018270767e-08 1.7671539284228046e-08 0.9999999999999998 1.5708000430119111
        children [
          USE WHEEL
        ]
        name "solid(2)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL4 HingeJoint {
      jointParameters HingeJointParameters {
        position -2.6895122773572173e-17
        axis 0 0 1
        anchor -0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint Solid {
        translation -0.060001096839924606 7.615246225842571e-07 -0.05000000839401549
        rotation -1.7610008372639072e-08 1.7610180209133872e-08 0.9999999999999997 1.570799956989706
        children [
          USE WHEEL
        ]
        name "solid(3)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF POLE HingeJoint {
      jointParameters HingeJointParameters {
        position -4.812689688184746e-16
        axis 0 0 1
        anchor 0 0.03000000000047226 0
        minStop -1.3
        maxStop 1.3
      }
      device [
        DEF POLE_POS_SENSOR PositionSensor {
          name "polePosSensor"
        }
      ]
      endPoint Solid {
        translation -0.00046553268549664073 0.5300037594690288 -5.369584240411803e-07
        rotation -0.5769874410964111 -0.5775313877690047 0.5775318077462255 2.0949379676602042
        children [
          DEF POLE_ENDPOINT Solid {
            translation 0.5000000000000002 0 0
            scale 0.1 0.1 0.1
            children [
              DEF BALL Shape {
                appearance PBRAppearance {
                  baseColor 0.03137254901960784 0.047058823529411764 1
                  metalness 0
                }
                geometry Sphere {
                  radius 0.1
                }
              }
            ]
            boundingObject USE BALL
            physics Physics {
              density -1
              mass 0.02
            }
          }
          DEF POLE Shape {
            appearance PBRAppearance {
              baseColor 1 0.1568627450980392 0.1568627450980392
              roughness 1
              metalness 0
            }
            geometry Box {
              size 1 0.020000000000000004 0.019999999999999997
            }
          }
        ]
        name "pole"
        boundingObject USE POLE
        physics Physics {
          density -1
          mass 0.2
        }
      }
    }
  ]
  boundingObject USE BODY
  physics Physics {
    density -1
    mass 2
  }
  controller "robotSupervisorController"
  supervisor TRUE
}
